pkg/_internal/policy/policy_test.go

```diff
@@ -9,55 +9,70 @@ import (
+	testutil "github.com/Kuadrant/multicluster-gateway-controller/test/util"
 )
 
 func TestGetTargetRefValueFromPolicy(t *testing.T) {
-	t.Run("should use target namespace", func(t *testing.T) {
-		policy := &v1alpha1.DNSPolicy{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "test-policy",
-				Namespace: "test-policy-ns",
-			},
-			Spec: v1alpha1.DNSPolicySpec{
-				TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
-					Group: "gateway.networking.k8s.io",
-					Kind:  "Gateway",
-					Name:  gatewayv1beta1.ObjectName("test-gateway"),
-				},
-			},
-		}
+	type args struct {
+		policy common.KuadrantPolicy
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "should use target namespace",
+			args: args{
+				policy: &v1alpha1.DNSPolicy{
 
-		res := GetTargetRefValueFromPolicy(policy)
-		expected := "test-gateway,test-policy-ns"
-		if res != expected {
-			t.Errorf("GetTargetRefValueFromPolicy returned %v; expected %v", res, expected)
-		}
-	})
-
-	t.Run("should use policy namespace when no target namespace set", func(t *testing.T) {
-		typedNamespace := gatewayv1beta1.Namespace("test-gateway-ns")
-		policy := &v1alpha1.DNSPolicy{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "test-policy",
-				Namespace: "test-policy-ns",
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-policy",
+						Namespace: "test-policy-ns",
+					},
+					Spec: v1alpha1.DNSPolicySpec{
+						TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+							Group: "gateway.networking.k8s.io",
+							Kind:  "Gateway",
+							Name:  "test-gateway",
+						},
+					},
+				},
 			},
-			Spec: v1alpha1.DNSPolicySpec{
-				TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
-					Group:     "gateway.networking.k8s.io",
-					Kind:      "Gateway",
-					Name:      gatewayv1beta1.ObjectName("test-gateway"),
-					Namespace: &typedNamespace,
+			want: "test-gateway,test-policy-ns",
+		},
+		{
+			name: "should use policy namespace when no target namespace set",
+			args: args{
+				policy: &v1alpha1.DNSPolicy{
+					TypeMeta: metav1.TypeMeta{},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-policy",
+						Namespace: "test-policy-ns",
+					},
+					Spec: v1alpha1.DNSPolicySpec{
+						TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+							Group:     "gateway.networking.k8s.io",
+							Kind:      "Gateway",
+							Name:      "test-gateway",
+							Namespace: testutil.Pointer(gatewayv1beta1.Namespace("test-gateway-ns")),
+						},
+					},
 				},
 			},
-		}
-
-		res := GetTargetRefValueFromPolicy(policy)
-		expected := "test-gateway,test-gateway-ns"
-		if res != expected {
-			t.Errorf("GetTargetRefValueFromPolicy returned %v; expected %v", res, expected)
-		}
-	})
+			want: "test-gateway,test-gateway-ns",
+		},
+	}
+	for _, tt := range tests {
```

Comment:
Looks like we're changing the t.Run approach in this test to a struct that is iterated over.
The struct approach seems to be the pattern in use elsewhere in the repo, so seems reasonable to make this change.